src/types.ts

```diff
@@ -0,0 +1,139 @@
+import { act as RTRAct } from 'react-test-renderer'
```

Comment:
I'm not sure we can have this import here as we can rely on them having this dependency installed and TS being statically analysed, it's going to straight up fail for consumers without it.

It sucks, but we might need to just define our own interface for `act` and let the duck typing take care of it. Thoughts?

Reply:
Yeah I wasn't sure about this. We could make our own wrapper similar to how it looks in `server/pure` –

```ts
export type ServerModifiedAct = (
  cb: ServerActCallbackAsync | ServerActCallback
) => Promise<undefined> | void
```

I'm also concerned how the type will work for `createRenderHook` in `core/index` when it's built whether it will correctly assert if you're using `react-hooks-testing-library/server` you **will** get hydrate in the return. Probably something to test in beta.